src/script/common/c_content.cpp

```diff
@@ -669,6 +740,400 @@ ContentFeatures read_content_features(lua_State *L, int index)
 	return f;
 }
 
+void push_content_features(lua_State *L, const ContentFeatures &c)
+{
+	std::string paramtype = "none";
+	std::string paramtype2 = "none";
+	std::string drawtype = "normal";
+	std::string liquid_type = "none";
+	
+	// paramtype
+	switch (c.param_type)
+	{
+		case CPT_NONE:
+		{
+			// paramtype already equals "none"
+			break;
+		}
+		case CPT_LIGHT:
+		{
+			paramtype = "light";
+			break;
+		}
+	}
+	
+	// paramtype2
+	switch (c.param_type_2)
+	{
+		case CPT2_NONE:
+		{
+			// paramtype2 already equals "none"
+			break;
+		}
+		case CPT2_FULL:
```

Comment:
Please stop the code duplication at this point and reuse the lookup table `ScriptApiNode::es_ContentParamType2` by moving to `nodedef`. (Same for the drawtypes)